src/compiler/checker.ts

```diff
@@ -5961,6 +5961,17 @@ namespace ts {
                         return setOriginalNode(typeToTypeNodeHelper(getTypeFromTypeNode(node), context), node);
                     }
                     if (isLiteralImportTypeNode(node)) {
+                        if (isInJSDoc(node) &&
+                            getNodeLinks(node).resolvedSymbol &&
+                            (
+                                // The import type resolved using jsdoc fallback logic
+                                (!node.isTypeOf && !(getNodeLinks(node).resolvedSymbol!.flags & SymbolFlags.Type)) ||
+                                // The import type had type arguments autofilled by js fallback logic

```

Comment:
where is this tested?

Reply:
I didn't add one; but this case is copied from the same case for type references - I just added a comment explaining it, since the condition was getting hairy.